src/main/java/com/google/cloud/flink/bigquery/BigQueryDynamicTableFactory.java

```diff
@@ -203,9 +218,20 @@ private LinkedList<String> getReadStreamNames(ReadableConfig options) {
       ReadSession readSession =
           BigQueryReadSession.getReadsession(credentials, bqconfig, bigQueryReadClientFactory);
       List<ReadStream> readsessionList = readSession.getStreamsList();
-      ArrowFormatFactory.setSchema(readSession.getArrowSchema());
-      ArrowFormatFactory.setSelectedFields(bqconfig.getSelectedFields());
-      // Iterate over read stream
+      Schema arrowReadSchema =
+          MessageSerializer.deserializeSchema(
+              new ReadChannel(
+                  new ByteArrayReadableSeekableByteChannel(
+                      readSession.getArrowSchema().getSerializedSchema().toByteArray())));
+      this.arrowReadSessionSchema = arrowReadSchema.toJson();
+      String fieldList = new String();
+      for (int i = 0; i < arrowReadSchema.getFields().size(); i++) {
```

Comment:
This will leave a redundant `,` at the end. Why not just call:
```
String fieldList = arrowReadSchema.getFields().stream().map(Field::getName).collect(Collectors.joining(","));
```

Reply:
Changed to stream